src/backend/catalog/index.c

```diff
@@ -2471,7 +2473,8 @@ IndexBuildScan(Relation parentRelation,
 	{
 		snapshot = SnapshotAny;
 		/* okay to ignore lazy VACUUMs here */
-		OldestXmin = GetOldestXmin(parentRelation, true);
+		if (!RelationIsAppendOptimized(parentRelation))
+			OldestXmin = GetOldestXmin(parentRelation, true);
```

Comment:
Comment `okay to ignore lazy VACUUMs here` have to be assigned to line where `GetOldestXmin()` is called.

Please add in the comment above if statement as TODO that in future there is planned to add support for AO tables to cutt off definitely dead tuples using `OldestXmin` value in the index building routine.

Reply:
No public link. This the same functionality as using NonVacuumableSnapshot and have to be added to fix corruption issues of vacuum.

Reply:
Despite it's a good idea it may or may not be included here. It's OK to mention in a PR body, that we have some plans over NonVacuumableSnapshot with it's reasoning(pros and cons), but we shouldn't place such info here, IMO.

Reply:
> but we shouldn't place such info here, IMO.

OK, then simply add TODO: `add support to cut off definitely dead rows using OldestXmin for AO tables`